test/RequestQueueTests.js

```diff
@@ -1,165 +1,300 @@
-var nock = require( 'nock' );
-var superagent = require( 'superagent' );
-var should = require( 'should' );
-var sinon = require( 'sinon' );
+var superagent = require( 'superagent' ),

```

Comment:
Tests were rewritten and fixed in the cases that they were doing false positives. Also added more in depth testing around the queuing logic as well as the retry logic